hedera-node/docs/design/services/token-service/airdrops/token-claim-airdrop.md

```diff
@@ -0,0 +1,165 @@
+                    - Then delegate it to `CryptoTransfer.handler()` to execute the transfer
+        - Token transfers and associations should be externalized using the `tokenTransferLists` and `automatic_token_associations` fields in the transaction record
+    - Fees calculation
+
+### Zero-Balance accounts
+
+An account with no open auto-association slots can receive airdrops but must send a `TokenClaimAirdrop` transaction, which requires a payer. If the account has zero hbars, then it can still claim the transfer if someone else is willing to pay for that transaction. For example, a Dapp could be the payer on the transaction. Both the Dapp and the account must sign the transaction.
+
+### Hollow accounts
+
+Any existing hollow accounts that were created before [HIP-904](https://hips.hedera.com/hip/hip-904) will have no or limited number of `maxAutoAssociations` depending on if they were created with HBAR or token transfer respectively. That means an airdrop of unassociated tokens to such accounts will result in a pending transfer.
+Performing `TokenClaimAirdrop` for such hollow account will also complete the account by setting its key which will obtained from the required transaction signature. Completing the hollow account should not modify the `maxAutoAssociations` on the account. That should always be an explicit step by a user.
+
+## Acceptance Tests
+
+All of the expected behaviour described below should be present only if the new `TokenClaimAirdrop` feature flag is enabled.
+
+- Given existing pending airdrop in state when valid `TokenClaimAirdrop` transaction containing entry for the same pending airdrop is performed then the `TokenClaimAirdrop` should succeed resulting in:
+    - the tokens being claimed should be automatically associated with the `receiver_id` account
+    - the tokens being claimed should be transferred to the `receiver_id` account
+    - the pending airdrop should be removed from state
+- Given a successful `TokenClaimAirdrop`  transaction having a hollow account as `receiver_id` should also complete the account without modifying its `maxAutoAssociations` value
+- Given successful `TokenClaimAirdrop` when another `TokenClaimAirdrop` for the same airdrop is performed then the second `TokenClaimAirdrop` should fail
+- `TokenClaimAirdrop` transaction with no pending airdrops entries should fail
+- `TokenClaimAirdrop` transaction with more than 10 pending airdrops entries should fail
+- `TokenClaimAirdrop` transaction containing duplicate entries should fail
+- `TokenClaimAirdrop` transaction containing pending airdrops entries which do not exist in state should fail
+- `TokenClaimAirdrop` transaction not signed by the account referenced by a `receiver_id` for each entry in the pending airdrops list should fail
+- `TokenClaimAirdrop` transaction with a `sender_id` account that does not have sufficient balance of the claimed token should fail
```

Comment:
Please add a test that checks the behavior when the feature flag is disabled.

Reply:
Addressed in https://github.com/hashgraph/hedera-services/pull/13206.